client/client.go

```diff
@@ -352,24 +352,24 @@ func (m *K8sClient) RolloutStatefulSets(namespace string) error {
 		// wait for the rollout to be complete
 		scmd := fmt.Sprintf("kubectl rollout status statefulset %s --namespace %s", s.Name, namespace)
 		log.Info().Str("Command", scmd).Msg("Waiting for StatefulSet rollout to finish")
-		if err := ExecCmd(scmd); err != nil {
+		if err := ExecCmdWithContext(ctx, scmd); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
 // RolloutRestartBySelector rollouts and restarts object by selector
-func (m *K8sClient) RolloutRestartBySelector(namespace string, resource string, selector string) error {
+func (m *K8sClient) RolloutRestartBySelector(namespace string, resource string, selector string, ctx context.Context) error {
```

Comment:
Nit: [Official docs](https://pkg.go.dev/context) recommend always using `ctx` as the first param in a function call. Should probably stick to this for the sake of convention.